Main/ClientApp/src/Components/Common/MessageWithLoader.tsx

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { Grid, Typography } from "@material-ui/core";
+import Loader from "react-loader-spinner";
+import Center from "./Center";
+
+type MessageWithLoaderProps = {
+  text: string;
+};
+
+export default function MessageWithLoader(props: MessageWithLoaderProps) {
+  return (
+    <Center>
+      <Grid
+        container
+        justify="center"
+        //   alignItems="center"
+        spacing={2}
+      >
+        <Grid item>
+          <Typography variant="h3" align="center" style={{ color: "grey" }}>
+            {props.text}
+          </Typography>
+        </Grid>
+        <Grid item>
```

Comment:
Seems like only this grid item is different from Message component, the other lines are exactly the same and copied. Consider adding a prop flag in Message that will decide whether to render the Loader or not.